Add docker

Sweta304 opened the pull request.

Sweta304 committed:
refactor:
- приложение разделено на два модуля
- добавлена возможность работы с docker

Sweta304 committed:
удалены неиспользуемые импорты и закомменченый код

Sweta304 committed:
fix:
- исправлены тесты
refactor:
- удалена аннотация @Data

Sweta304 committed:
.

Sweta304 committed:
fix:
изменены application.properties для тестов

Sweta304 committed:
fix:
добавлены методы equals() и hashcode()

alexache commented on docker-compose.yaml:
```diff
@@ -0,0 +1,39 @@
+    ports:
+      - "8080:8080"
+    depends_on:
+      - db
```
Вроде бы по логике наоборот: сервис зависит от базы данных, а гейтвей от сервиса

alexache commented on gateway/pom.xml:
```diff
@@ -0,0 +1,80 @@
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.data</groupId>
```
Зачем зависимость от базы данных гейтвею?

alexache commented on gateway/src/main/resources/application.properties:
```diff
@@ -0,0 +1,5 @@
+logging.level.org.springframework.web.client.RestTemplate=DEBUG
+#logging.level.org.apache.http=DEBUG
```
Комментарии в пропертях тоже нужно подчишать

alexache commented on server/pom.xml:
```diff
@@ -0,0 +1,96 @@
+        </dependency>
+        <dependency>
+            <groupId>jakarta.validation</groupId>
+            <artifactId>jakarta.validation-api</artifactId>
```
Валидации теперь должны быть только в гейтвее

alexache commented on server/src/main/java/ru/practicum/shareit/booking/BookingMapper.java:
```diff
@@ -11,7 +12,8 @@
 import static ru.practicum.shareit.user.UserMapper.toUserDto;
 
-@Data
+@Getter
```
А для чего эти аннотации здесь? У класса нет пропертей

alexache commented on server/src/main/java/ru/practicum/shareit/booking/model/Booking.java:
```diff
@@ -33,4 +36,17 @@ public class Booking {
     private BookStatus status;
+
+    @Override
+    public boolean equals(Object o) {
```
Методы equals/hashcode очень сложно писать правильно для JPA моделей. Лучше не писать. Есть ли в них необходимость?

Sweta304 replied:
hashcode нет, а без equals валятся тесты(

Sweta304 replied:
нет, в моделях действительно нет необходимости этих методов, удалила

alexache commented on server/src/main/java/ru/practicum/shareit/booking/service/BookingServiceImpl.java:
```diff
@@ -95,18 +98,6 @@ public BookingDto getBookingById(Long bookingId, Long owner) throws IncorrectOwn
-            }
-        } catch (IllegalArgumentException e) {
-            throw new IncorrectBookingStatusException("некорректный статус бронирования");
-        }
```
validatePagination тоже должно быть только в гейтвее

alexache commented on server/src/main/java/ru/practicum/shareit/item/CommentMapper.java:
```diff
@@ -1,12 +1,14 @@
 import java.time.LocalDateTime;
 
-@Data
+@Getter
```
Такой эе коммент, как и в другом маппере

alexache commented on server/src/main/java/ru/practicum/shareit/item/ItemMapper.java:
```diff
@@ -11,7 +12,8 @@
 import java.util.List;
 
-@Data
+@Getter
```
И здесь

alexache commented on server/src/main/java/ru/practicum/shareit/item/dto/ItemDto.java:
```diff
@@ -0,0 +1,46 @@
+    private Long id;
+    private Long requestId;
+
+    public static boolean validateItem(ItemDto itemDto) {
```
Валдации должны быть только в гейтвее

alexache commented on server/src/main/java/ru/practicum/shareit/item/model/Comment.java:
```diff
@@ -25,4 +28,17 @@ public class Comment {
     private User author;
+
+    @Override
+    public boolean equals(Object o) {
```
Повторный коммент про эти методы

alexache commented on server/src/main/java/ru/practicum/shareit/item/model/Item.java:
```diff
@@ -30,4 +30,17 @@ public class Item {
     private ItemRequest request;
+
+    @Override
+    public boolean equals(Object o) {
```
Повторный коммент

alexache commented on server/src/main/java/ru/practicum/shareit/requests/dto/ItemRequestDto.java:
```diff
@@ -0,0 +1,38 @@
+@NoArgsConstructor
+public class ItemRequestDto {
+    private Long id;
+    @NotNull
```
Валидации должны быть только в гейтвее

alexache commented on server/src/main/java/ru/practicum/shareit/requests/model/ItemRequest.java:
```diff
@@ -30,4 +33,17 @@ public class ItemRequest {
     private List<Item> responses;
+
+    @Override
+    public boolean equals(Object o) {
```
И еще один коммент про эти методы

alexache commented on server/src/main/java/ru/practicum/shareit/user/UserMapper.java:
```diff
@@ -1,10 +1,12 @@
 import ru.practicum.shareit.user.model.User;
 
-@Data
+@Getter
```
Еще один повторный коммент

alexache commented on server/src/main/java/ru/practicum/shareit/user/model/User.java:
```diff
@@ -11,7 +12,8 @@
 @Table(name = "users")
-@Data
+@Getter
+@Setter
```
И здесь equals/hashcode ниже лишние

Sweta304 committed:
fix:
- удалены equals() и hashcode() из JPA моделей
- исправлен docker-compose.yaml
- удалены лишние зависимости из POM файлов

Sweta304 committed:
fix:
- пропущенная валидация перенесена в gateway

Sweta304 committed:
.

Sweta304 committed:
.

alexache commented on server/src/main/java/ru/practicum/shareit/item/service/ItemServiceImpl.java:
```diff
@@ -107,45 +101,34 @@ public ItemWithBooking getItem(Long itemId, Long owner) throws ItemNotFoundExcep
+        Sort sortByCreated = Sort.by(Sort.Direction.ASC, "id");
+        Pageable page = new MyPageable(from, size, sortByCreated);
+        Page<Item> requestPage = itemRepository.findAll(page);
+        return requestPage.getContent()
```
Как-то здесь я пропустил в прошлый раз - этот поиск нужно делать с помощью SQL запроса, а не на уровне обработки коллекции

alexache commented on server/src/test/java/ru/practicum/shareit/user/UserServiceTest.java:
```diff
@@ -52,7 +53,7 @@ void getAllUsers() {
+        when(userRepository.save(any())).thenReturn(user);
         UserDto userDtoSample = userService.addUser(toUserDto(user));
 
         assertNotNull(userDto);
```
Зачем вообще проверять эту переменную?
